components/channel_invite_modal/index.js

```diff
@@ -6,16 +6,34 @@ import {bindActionCreators} from 'redux';
 
 import {getTeamStats} from 'mattermost-redux/actions/teams';
 import {getProfilesNotInChannel, searchProfiles} from 'mattermost-redux/actions/users';
-import {getProfilesNotInCurrentChannel, getProfilesNotInCurrentTeam} from 'mattermost-redux/selectors/entities/users';
+import {getProfilesNotInCurrentChannel, getProfilesNotInCurrentTeam, getProfilesNotInTeam, makeGetProfilesNotInChannel} from 'mattermost-redux/selectors/entities/users';
 
 import {addUsersToChannel} from 'actions/channel_actions';
 
 import ChannelInviteModal from './channel_invite_modal.jsx';
 
-function mapStateToProps(state) {
-    return {
-        profilesNotInCurrentChannel: getProfilesNotInCurrentChannel(state),
-        profilesNotInCurrentTeam: getProfilesNotInCurrentTeam(state),
+function makeMapStateToProps(initialState, initialProps) {
+    let doGetProfilesNotInChannel;
+    if (initialProps.channelId && initialProps.teamId) {
+        doGetProfilesNotInChannel = makeGetProfilesNotInChannel();
+    }
+
+    return function mapStateToProps(state, props) {
```

Comment:
Non blocking, but could this be an arrow function? That's usually the standard we go with for in-line functions.